### PR TITLE
Fix field guide, metadata, help and tutorial for dark theme

### LINKED ISSUE #643: Support light/dark theme for field guide
**Package** lib-classifier

**Feature or Issue Description**
Add light/dark theme support.

### PR DESCRIPTION
**Pull request by mcbouslog**
Package: lib-classifier, lib-react-components

Closes #643 

Describe your changes:
- temporary solution until issue related to React or Grommet can be sorted
- passes updated theme as prop to Modal component
- adds `<Grommet theme={theme} />` to Modal component

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/Banners/components/Banner/Banner.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ class Banner extends Component {
                 <Box direction='column' margin='xsmall'>
                   <Triangle />
                   <Box
-                    background={mode === 'light' ? 'white' : 'dark-2'}
+                    background={dark ? 'dark-2' : 'white'}
```
Comment (rogerhutchings) [inline packages/lib-classifier/src/components/Classifier/components/Banners/components/Banner/Banner.js]: You should also be able to do this:

```
background={{
  dark: 'dark-2',
  light: 'white
}}
```

Comment (rogerhutchings) [reply packages/lib-classifier/src/components/Classifier/components/Banners/components/Banner/Banner.js]: ...which may mean you can remove `withTheme` and related stuff!

File: packages/lib-classifier/src/components/Classifier/components/Banners/components/Banner/components/Triangle/Triangle.js [hunk 2]
```diff
@@ -38,8 +37,20 @@ Triangle.propTypes = {
...
     }),
-    mode: oneOf(['dark', 'light'])
+    dark: bool
   })
 }
 
+Triangle.defaultProps = {
+  theme: {
+    dark: false,
+    global: {
```
Comment (rogerhutchings) [inline packages/lib-classifier/src/components/Classifier/components/Banners/components/Banner/components/Triangle/Triangle.js]: I don't think you need to add `global` as a default prop, just `dark`

File: packages/lib-react-components/src/Modal/Modal.js [hunk 3]
```diff
@@ -1,18 +1,19 @@
 import PropTypes from 'prop-types'
 import React from 'react'
+import { Grommet } from 'grommet'
 
 import WithLayer from './WithLayer'
 import ModalBody from './components/ModalBody'
 import ModalHeading from './components/ModalHeading'
 
-function Modal ({ children, className, closeFn, pad, title }) {
+function Modal ({ children, className, closeFn, pad, theme, title }) {
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/Modal/Modal.js]: If you add `withTheme` to this component, you don't need to pass it in anywhere else 💯

File: packages/lib-react-components/src/Modal/Modal.js [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
   return (
-    <React.Fragment>
+    <Grommet theme={theme}>
       <ModalHeading className={className} closeFn={closeFn} title={title} />
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/Modal/Modal.js]: I don't think `className`s are needed here, as we're not wrapping anything in `styled()`

**Comment by mcbouslog** [comment]
Yes originally intended to use `withTheme` but there's an issue with either Grommet or React, I hope this codesandbox illustrates well - https://codesandbox.io/s/6jnn5ornkn. If you toggle to dark mode, the theme that the Layer children components get doesn't have `dark: true` like they should.

In process of cleaning up codesandbox to further a discussion in Grommet Slack about the issue, I think if the theme has a Layer property that includes a dark and light color setting, even if it's to the same color, somehow that updates the theme as expected.

**Comment by mcbouslog** [comment]
Closing this in favor of #825

**Pull request closed without merging**